xbmc/filesystem/MusicDatabaseDirectory/DirectoryNodeAlbumRecentlyPlayed.cpp

```diff
@@ -24,9 +24,9 @@ CDirectoryNodeAlbumRecentlyPlayed::CDirectoryNodeAlbumRecentlyPlayed(const std::
 NODE_TYPE CDirectoryNodeAlbumRecentlyPlayed::GetChildType() const
 {
   if (GetName()=="-1")
-    return NODE_TYPE_ALBUM_RECENTLY_PLAYED_SONGS;
+    return NODE_TYPE_DISC;
```

Comment:
There are no recently played songs to get to, only albums. Returning NODE_TYPE_DISCS here ultimately will give you (all) the songs on any recently played album which is the same thing isn't it ?  There doesn't appear any way currently to actually navigate to a list of recently played songs without using a smart playlist.

Reply:
>There are no recently played songs to get to, only albums

You have to have played a song to have played (part of) an album, so I don't understand. Am I missing something? My concern is never reaching `CDirectoryNodeAlbumRecentlyPlayedSong` and how the contents of  `NODE_TYPE_ALBUM_RECENTLY_PLAYED_SONGS` node type is populated

Reply:
If you play a song then, unless it is a single (which I admit I have only just thought about), the album it is on becomes part of `recentlyplayedalbums`. If you then get that particular album then the song you played will be a part of it.  
I think that `GetRecentlyPlayedAlbumSongs` is un-needed because it doesn't show anything that you can't get to via `GetRecentlyPlayedAlbums` and then getting the songs for those albums. It's the same list ultimately, just the first one produces a list of tracks but it's not possible to navigate directly to that anyway. Perhaps I'm missing something, but as far as I can see, GRPAS just gets the songs off the recentlyplayedalbums, so showing the albums and then just `GetSongsByWhere` achieves the same thing - unless there is a way to navigate directly to that list without using a smart playlist ?

Reply:
Remember it is what happens when you click on "*all albums" item on the RecentlyPlayedAlbums node that we are discussing here. Behaviour before this PR would be to see all the songs from those X of the most recently played albums. Those albums are not fetched using criteria passed to GetAlbumsByWhere, hence there are no criteria to pass along to GetSongsFullByWhere. Instead a unique method/query (GetRecentlyPlayedAlbums) is used to list the albums, and another (GetRecentlyPlayedAlbumSongs) to fetch the songs from those X albums. 

We can click on an album and go to the discs (then songs) _of that album_, but we can't go to the songs of all those X albums via discs. I'm on another branch at the moment (and compile takes 30mins so not swapping) so can't try,  but I think that what you have would often end up  showing all songs